modules/openapi-generator/src/main/resources/typescript-fetch/recordGeneric.mustache

```diff
@@ -0,0 +1,148 @@
+};
+
+export type {{classname}}RecordPropsType = typeof {{classname}}RecordProps;
+export const {{classname}}Record = Record({{classname}}RecordProps, {{classname}}RecordProps.recType);
+export type {{classname}}Record = RecordOf<{{classname}}RecordPropsType>;
+
+knownRecordFactories.set({{classname}}RecordProps.recType, {{classname}}Record);
+
+{{#isEntity}}
+export const {{classname}}RecordEntityProps = {
+	...{{classname}}RecordProps,
+	recType: "{{classname}}RecordEntity" as "{{classname}}RecordEntity",
+{{#vars}}
+    {{#isEntity}}
+    {{name}}: {{#required}}"-1"{{/required}}{{^required}}null as string | null{{/required}},
+    {{/isEntity}}
+    {{#itemsAreEntities}}
+    {{name}}: {{#required}}List<string>(){{/required}}{{^required}}null as List<string> | null{{/required}},
+    {{/itemsAreEntities}}
+{{/vars}}
+};
+
+export type {{classname}}RecordEntityPropsType = typeof {{classname}}RecordEntityProps;
+export const {{classname}}RecordEntity = Record({{classname}}RecordEntityProps, {{classname}}RecordEntityProps.recType);
+export type {{classname}}RecordEntity = RecordOf<{{classname}}RecordEntityPropsType>;
+
+knownRecordFactories.set({{classname}}RecordEntityProps.recType, {{classname}}RecordEntity);
+{{/isEntity}}
+
+class {{classname}}RecordUtils extends ApiRecordUtils<{{classname}}, {{classname}}Record> {
+	public normalize(apiObject: {{classname}}, asEntity?: boolean): {{classname}} {
+		(apiObject as any).recType = {{#isEntity}}asEntity ? "{{classname}}RecordEntity" : {{/isEntity}}"{{classname}}Record";
+{{#vars}}
+        {{#isUniqueId}}
+        {{^required}}if (apiObject.{{name}}) { {{/required}}(apiObject as any).{{name}} = apiObject.{{name}}.toString();{{^required}} } {{/required}}
+        {{/isUniqueId}}
+        {{#isModel}}
+        {{^required}}if (apiObject.{{name}}) { {{/required}}{{#lambda.camelcase}}{{dataTypeAlternate}}{{/lambda.camelcase}}Utils.normalize(apiObject.{{name}});{{^required}} } {{/required}}
+        {{/isModel}}
+        {{#itemsAreModels}}
+        {{^required}}if (apiObject.{{name}}) { {{/required}}{{#lambda.camelcase}}{{itemsDataType}}{{/lambda.camelcase}}Utils.normalizeArray(apiObject.{{name}});{{^required}} } {{/required}}
+        {{/itemsAreModels}}
+{{/vars}}
+		return apiObject;
+	}
+{{#isEntity}}
+
+	public getSchema(): Schema {
+	    return new schema.Entity("{{#lambda.camelcase}}{{classname}}{{/lambda.camelcase}}", {
+{{#vars}}
+            {{#isEntity}}
+            {{#lambda.camelcase}}{{name}}{{/lambda.camelcase}}: {{#lambda.camelcase}}{{dataTypeAlternate}}{{/lambda.camelcase}}Utils.getSchema(),
+            {{/isEntity}}
+            {{#itemsAreEntities}}
+            {{#lambda.camelcase}}{{name}}{{/lambda.camelcase}}: [{{#lambda.camelcase}}{{itemsDataType}}{{/lambda.camelcase}}Utils.getSchema()],
+            {{/itemsAreEntities}}
+{{/vars}}
+		});
+	}
+{{/isEntity}}
+{{#isMetaDataResponse}}
+{{#vars.1}}
+
+	public fromApi_data(apiObject: {{classname}}): {{{dataTypeAlternate}}} {
```

Comment:
like I said above, all this thing with the dataTypeAlternate must be  handled in the codegen code